SwiftWisdomTests/CommonTypes/MulticastTests.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  MulticastTests.swift
+//  SwiftWisdom
+//
+//  Created by Son Le on 2/9/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
+import XCTest
+@testable import SwiftWisdom
+
+@objc private protocol BroadcasterDelegate: class {
+    func broadcaster(_ broadcaster: Broadcaster, broadcasted message: String)
+}
+
+private class Broadcaster: NSObject, Multicast {
+
+    typealias Delegate = BroadcasterDelegate
+    var delegateReferences = [Weak<BroadcasterDelegate>]()
+
+    func broadcast(that message: String) {
+        for delegate in delegates {
+            delegate.broadcaster(self, broadcasted: message)
+        }
+    }
+}
+
+private class Subscriber: BroadcasterDelegate {
+
+    let closure: (String) -> Void
+
+    init(_ closure: @escaping (String) -> Void) {
+        self.closure = closure
+    }
+
+    func broadcaster(_ broadcaster: Broadcaster, broadcasted message: String) {
+        closure(message)
+    }
+}
+
+class MulticastTests: XCTestCase {
+
+    func testAddingDelegates() {
+        var messageRecipients = [String]()
+        let subscriberA = Subscriber { message in
+            if message == "We are intrepid" {
+                messageRecipients.append("A")
+            }
+        }
+        let subscriberB = Subscriber { message in
+            if message == "We are intrepid" {
+                messageRecipients.append("B")
+            }
+        }
+        let subscriberC = Subscriber { message in
+            if message == "We are intrepid" {
+                messageRecipients.append("C")
+            }
+        }
+
+        let broadcaster = Broadcaster()
+        broadcaster.add(delegate: subscriberA)
+        broadcaster.add(delegate: subscriberB)
+        broadcaster.add(delegate: subscriberC)
+        broadcaster.add(delegate: subscriberA)
```

Comment:
I would also suggest that this particular test be broken out - this test should be named `testAddOnSameDelegateShouldNotReadd`. 

The overall test should be named `testBroadcastShouldBroadcastToAllDelegates`